### PR TITLE
PHP 8 compatibility

### PR DESCRIPTION
**Pull request by Glancu**

### CODE REVIEW

**Comment by garak** [comment]
As you can see in travis logs, this is currently not working.
We need 2 additional steps:
1. prophecy php 8 support (see [related issue](https://github.com/phpspec/prophecy/issues/491))
2. allowing latest prophecy version (only after point 1 is solved)

**Comment by derrabus** [comment]
> As you can see in travis logs, this is currently not working.

If you want to to wait for external projects to adjust their php constraints, you could temporarily add the following line to the `nightly` run:

```
composer config platform.php 7.4.99
```

This is how Symfony currently runs the tests on php 8.

**Comment by Glancu** [comment]
> > As you can see in travis logs, this is currently not working.
> 
> If you want to to wait for external projects to adjust their php constraints, you could temporarily add the following line to the `nightly` run:
> 
> ```
> composer config platform.php 7.4.99
> ```
> 
> This is how Symfony currently runs the tests on php 8.

I added commit with script platform, but changes in PR not detected. Do you know why?
https://github.com/Glancu/KnpMenu/commit/23025d3fbf6582790281c2c47a8d3867f1dff0c1

**Comment by derrabus** [comment]
The php 8 build is deep red now because PHPUnit 8.3 (installed by PhpUnitBridge) is incompatible with php 8. Symfony is struggling with the same issue at the moment, see #37564.

We might be able to fix this by dropping PhpUnitBridge and install PHPUnit 9.3-dev for the nightly run. 🤔

File: composer.json [hunk 1]
```diff
@@ -48,5 +48,6 @@
         "branch-alias": {
             "dev-master": "3.2-dev"
         }
-    }
+    },
```
Comment (garak) [inline composer.json:51]: like previously noted, this should go in travis configuration

Comment (Glancu) [reply composer.json]: Fixed

**Comment by stof** [comment]
> We might be able to fix this by dropping PhpUnitBridge and install PHPUnit 9.3-dev for the nightly run.

Or forcing the PHPUnit version installed by the bridge in the PHP 8 job for now.

**Comment by derrabus** [comment]
> Or forcing the PHPUnit version installed by the bridge in the PHP 8 job for now.

PHPUnit 9.3 breaks the bridge atm, I'm afraid.

**Pull request merged**